light/store/db/db.go

```diff
@@ -169,13 +171,8 @@ func (s *dbs) FirstLightBlockHeight() (int64, error) {
 	}
 	defer itr.Close()
 
-	for itr.Valid() {
-		key := itr.Key()
-		_, height, ok := parseLbKey(key)
-		if ok {
-			return height, nil
-		}
-		itr.Next()
+	if itr.Valid() {
```

Comment:
Implementing this for the other PR. This should return the height if there is no error. Otherwise, there might be a reason for the error where we should keep iterating. In the new layout I think it is these two conditions

```	if lightBlockPrefix != prefixLightBlock {
		err = fmt.Errorf("expected light block prefix but got: %d", lightBlockPrefix)
	}
	if dbPrefix != storePrefix {
		err = fmt.Errorf("parsed key has a different prefix. Expected: %s, got: %s", storePrefix, dbPrefix)
	}
	
	
``` 

And in the old one it is the first condition when there is no submatch.